src/lib/quoter.js

```diff
@@ -1,17 +1,88 @@
 'use strict'
 
 const ILQP = require('ilp').ILQP
+const LiquidityCurve = require('ilp-routing').LiquidityCurve
 const InvalidAmountSpecifiedError = require('../errors/invalid-amount-specified-error')
 const IlpError = require('../errors/ilp-error')
+const log = require('../common/log').create('quoter')
 
 class Quoter {
-  constructor (ledgers) {
+  /**
+   * @param {Ledgers} ledgers
+   * @param {Object} config
+   * @param {Integer} config.quoteExpiry
+   */
+  constructor (ledgers, config) {
     this.ledgers = ledgers
     this.tables = ledgers.tables
+    this.quoteExpiryDuration = config.quoteExpiry // milliseconds
+  }
+
+  /**
+   * @param {Object} request
+   * @param {IlpAddress} request.sourceAccount
+   * @param {IlpAddress} request.destinationAccount
+   * @param {Integer} request.destinationHoldDuration
+   * @param {Boolean} [request._checkCache] default: false
+   * @param {Boolean} [request._shiftCurve] default: true
+   * @returns {Object}
+   */
+  * quoteLiquidity (request) {
+    const liquidityQuote = yield this._quoteLiquidity(
+      Object.assign({_shiftCurve: true}, request))
+    if (!liquidityQuote) return null
+    return Object.assign({}, liquidityQuote, {
+      liquidityCurve: liquidityQuote.liquidityCurve.toBuffer(),
+      expiresAt: new Date(liquidityQuote.expiresAt)
+    })
+  }
+
+  * _quoteLiquidity (request) {
+    const hop = this.tables.localTables.findBestHopForSourceAmount(request.sourceAccount, request.destinationAccount, '0')
+    if (!hop) return Promise.resolve(null)
+    const connector = hop.bestHop
+    const fullRoute = hop.bestRoute
+    const shiftBy = request._shiftCurve
+      ? this.tables.getScaleAdjustment(this.ledgers, fullRoute.sourceLedger, fullRoute.nextLedger)
+      : 0
+    if (isCurveExpired(fullRoute)) {
+      const headRoute = this.tables.localTables.getLocalRoute(fullRoute.sourceLedger, fullRoute.nextLedger)
+      const tailQuote = yield ILQP.quoteByConnector({
+        plugin: this.ledgers.getPlugin(fullRoute.nextLedger),
+        connector,
+        quoteQuery: {
+          destinationAccount: request.destinationAccount,
+          destinationHoldDuration: request.destinationHoldDuration
+        }
+      })
+      if (tailQuote.code) throw new IlpError(tailQuote)
```

Comment:
isn't there a cleaner way to check if the response is an error? something like `(tailQuote.responseType === ERROR)` instead of `(tailQuote.code)`?